### ISSUE TITLE
Do you want auto releases?

### ISSUE THREAD

**Issue by JLLeitschuh** [dan200/ComputerCraft#165]

**Comment by apemanzilla** [comment]
To be honest I don't see the point. If you did a release for every single commit to master you'd end up with a *lot* of releases, and if users need to get a specific change that hasn't been released yet it's trivial to build the project from source.

**Comment by Wojbie** [comment]
True. But many people don't know how to use gradle or even anything about making cc mods.. Having an automated weekly build would make it more accessible to random people on internet so they can find bugs.

**Comment by apemanzilla** [comment]
Automated weekly builds make sense, but one for every commit would be excessive IMO.

**Comment by JLLeitschuh** [comment]
I can set it up to run daily, weekly, or monthly easily.
Or any variance of that, (you'd just check to see if the day is some modulo of some multiple and skip a release if it's not).

https://docs.travis-ci.com/user/cron-jobs/

Weekly makes sense.

**Comment by Wojbie** [comment]
It might sounds wired but does it skip the release if there was no commits since last one? Because there can be sometimes a week or 2 when project stalls cause people are working on stuff and making extra weekly one makes no sense then.

**Comment by oddstr13** [comment]
With Jenkins, you can set it up to check for changes in the git repo using crontab-like syntax;
![jenkins build triggers screenshot](https://thoas.feralhosting.com/oddstr13/sharex/image/firefox_2017-05-03_20-28-49.png)

**Comment by JLLeitschuh** [comment]
@dan200 I want your feedback before I try to implement this.

**Comment by CrazedProgrammer** [comment]
I made a cron job on my server to build and package a redistribution (with SquidDev's modified build.gradle) of the main repository every 3 hours (starting at 0:30).  
You can find it at https://crzd.me/files/ComputerCraft-nightly.zip
There is also a maven repository for it at https://crzd.me/files/maven/ (the xml is located at https://crzd.me/files/maven/dan200/computercraft/ComputerCraft/maven-metadata-local.xml).

I'm waiting on PR https://github.com/dan200/ComputerCraft/pull/184 to get merged, so I can also release this on the forums.

**Comment by JLLeitschuh** [comment]
I'd ideally like to see that the releases get published through the official channels:
- here on github through tagged releases
- also with whatever curse publishing that exist
- anywhere else @dan200 wants release jars published

**Comment by CrazedProgrammer** [comment]
Yeah I can see that being the optimal solution for releases.  
This redistribution is just so people can use the absolute latest build of CC.

**Comment by Restioson** [comment]
I don't know how ideal this is. If new PR's are merged but not tested, this could break things. If we made tests and built if all tests passed, that could work... Nightly is excessive though. Once a week sounds good.

**Comment by vpontin** [comment]
@CrazedProgrammer these builds are for 1.9.4/1.10 or also works in 1.11.x?

**Comment by SquidDev** [comment]
@vpontin Just 1.9.4/1.10.2 at the moment. There is a 1.11.2 PR in the works.

**Comment by vpontin** [comment]
Oh thanks @SquidDev ! I'll stick with unborked until builds for 1.11 be possible

**Comment by CrazedProgrammer** [comment]
Hi guys.  
Yesterday I made a script that clones and (tries to) build every PR in this repo.  
You can see it at https://crzd.me/files/ComputerCraft-PRs/ though it is a mess of files. The jars are usually located at <pr>/ComputerCraft/build/libs.  
Not every build succeeds, and if this directory is empty then you'll be better off cloning and building it yourself.  
Hope this is useful!

**Comment by Restioson** [comment]
@CrazedProgrammer \*tests my pr*

**Comment by CrazedProgrammer** [comment]
Well, technically they are uncommonly downloaded :P

**Comment by Restioson** [comment]
True... 😛
